### DISCUSSION TITLE
How to pin a dependency version in a port manifest.

### DISCUSSION THREAD

**Discussion by crud89** [microsoft/vcpkg#23340] [Q&A] [answered]
I have created an overlay port for my project that depends on an older version of a dependency (fmt 7.1.3). Fmt 8 breaks the build of my port, so I have to revert back the vcpkg baseline. I attempted to do an overwrite in the port manifest to pin the old version, however overwrites are only supported in user manifests and appear to be ignored for ports. Is there any way to pin the version of the dependency in my port manifest?

I sure could simply update my project to use the newer version for fmt 8, however I want to make sure that the build still succeeds in the future, no matter which vcpkg baseline a user may have checked out.

**Comment by Osyotr** [comment, marked as answer]
Use overlay ports.

**Reply by crud89** [reply]
Sorry if this was not clear, but that's what I did for my project. Basically it is hosted in a registry that gets passed to vcpkg as an overlay port. Or did you mean putting the older baseline of fmt into my registry?

**Reply by Osyotr** [reply]
Sorry, I did not read your question thoroughly. My answer was for the case when _you_ want to use a port that requires older version of `fmt`. If you want to force users of this port to use older version of `fmt`, that is not possible.

**Reply by crud89** [reply]
Got it, thanks!
